test/testHelpers.hpp

```diff
@@ -439,6 +439,9 @@ bool noDoubleTests() {
     return ((isTypeDouble && !isDoubleSupported) ? true : false);
 }
 
+#define SUPPORTED_TYPE_CHECK(type) \
+    if (noDoubleTests<type>()) return;
+
```

Comment:
I see that we have the following
```cpp
#define SUPPORTED_TYPE_CHECK(type) \
  if (noDoubleTests<type>()) return; \
  if (noHalfTests<type>()) return
```
in PR #2413 , this will fail when the device has has no double support but half support - as in we will get false positives in test results, Unless I am missed seeing something in the changes.

I don't think reverse order will also work, because there could be devices with double support but no half support.

Reply:
Hm, is this okay for now because we only ammulate half computation using float for actual computation ?

Reply:
Nevermind, I missed the type match validation. My bad.